### PR TITLE
feature: Computing the nearest rail track from a point

### LINKED ISSUE #116: Computing the nearest rail track from a point
## Scenario summary
Currently the "RailTrack model" is not supported.

## Problem statement
Computing the distance from a rail track to a point and get the nearest

## Proposed solution
- [x] Create a RailTrack model
- [x] Write code in order to read these objects from a ShapeFile
- [x] Write code in order to compute the index
- [x] Compute distance from a point to these objects

### PR DESCRIPTION
**Pull request by fpalini**
## New features and improvements
- Created new OSM model called OSMRailTrack
- Created new OSM loader calles OSMRailwayLoader
- Added new maps files for nord-ovest railways in Italy

The files have been downloaded as *.osm.pbf from http://download.geofabrik.de/europe/italy.html.

The following commands have been used to filter railways (see https://wiki.openstreetmap.org/wiki/Key:railway) and compute the shapefiles:
```bash
osmium tags-filter --overwrite -o nord-ovest-rail.pbf nord-ovest-latest.osm.pbf nw/railway=light_rail,narrow_gauge,rail,disused 
ogr2ogr -f "ESRI Shapefile" -skip  nord-ovest-rail.shp nord-ovest-rail.pbf
```

The output files considered are the ones named *lines.**

## Breaking changes
Nothing

## Migration
Nothing

## Bug fixes
Nothing

## Related issue
Closes #116

### CODE REVIEW

File: src/main/scala/it/agilelab/gis/core/utils/ManagerUtils.scala [hunk 1]
```diff
@@ -12,7 +12,8 @@ object ManagerUtils {
       boundary: BoundaryPathGroup,
       postalCodes: Array[Path],
       roads: Array[Path],
-      houseNumbers: Array[Path]
+      houseNumbers: Array[Path],
+      railways: Array[Path]
```
Comment (andr3a87) [inline src/main/scala/it/agilelab/gis/core/utils/ManagerUtils.scala:16]: in this way, the feature of "raileways" is mandatory. or am I wrong?  in this case, it should be a feature that may not even be used in certain contexts

Comment (fpalini) [reply src/main/scala/it/agilelab/gis/core/utils/ManagerUtils.scala:16]: you are right, i considered it mandatory. I fix this behaviour

Comment (fpalini) [reply src/main/scala/it/agilelab/gis/core/utils/ManagerUtils.scala:16]: now the railways model path is optional, however I left that constructor as-is (the array can be eventually empty for railways). 

If the railways path is given, the IndexManager builds the index and the user can exploit the IndexSet which has the railways geometry list (as option).

File: pom.xml [hunk 2]
```diff
@@ -3,7 +3,7 @@
     <modelVersion>4.0.0</modelVersion>
     <groupId>it.agilelab</groupId>
     <artifactId>gis.lib</artifactId>
-    <version>1.5.0-SNAPSHOT</version>
+    <version>1.5.1-SNAPSHOT</version>
```
Comment (andr3a87) [inline pom.xml:6]: revert this.

**Pull request merged**